operator/pkg/model/ingestion/ingress.go

```diff
@@ -0,0 +1,199 @@
+	// TLS config with rules that match it.
+	// This is to approximate a set, keyed by hostname, so we can
+	// coalesce the config from a single Ingress.
+	// Coalescing the config from multiple Ingress resources is left for
+	// the transform component that takes a model and outputs CiliumEnvoyConfig
+	// or other resources.
+	insecureListenerMap := make(map[string]model.HTTPListener)
+
+	sourceResource := model.FullyQualifiedResource{
+		Name:      ing.Name,
+		Namespace: ing.Namespace,
+		Group:     "",
+		Version:   "v1",
+		Kind:      "Ingress",
+	}
+
+	if ing.Spec.DefaultBackend != nil {
+		// There's a default backend set up
+
+		// get the details for the default backend
+
+		backend := model.Backend{}
+		backend.Name = ing.Spec.DefaultBackend.Service.Name
+		backend.Namespace = ing.Namespace
+
+		backend.Port = &model.BackendPort{}
+
+		if ing.Spec.DefaultBackend.Service.Port.Name != "" {
+			backend.Port.Name = ing.Spec.DefaultBackend.Service.Port.Name
+		}
+
+		if ing.Spec.DefaultBackend.Service.Port.Number != 0 {
+			backend.Port.Port = uint32(ing.Spec.DefaultBackend.Service.Port.Number)
+		}
+
+		l := model.HTTPListener{
+			Hostname: "*",
+			Routes: []model.HTTPRoute{
+				{
+					Backends: []model.Backend{
+						backend,
+					},
+				}},
+			Port: 80,
+		}
+
+		l.Sources = model.AddSource(l.Sources, sourceResource)
+
+		insecureListenerMap["*"] = l
+	}
+
+	// Now, we range across the rules, adding them in as listeners.
+	for _, rule := range ing.Spec.Rules {
+
+		host := "*"
+
+		if rule.Host != "" {
+			host = rule.Host
+		}
+
+		l, ok := insecureListenerMap[host]
+		l.Port = 80
+		l.Sources = model.AddSource(l.Sources, sourceResource)
+		if !ok {
+			l.Name = fmt.Sprintf("ing-%s-%s-%s", ing.Name, ing.Namespace, host)
+		}
+
+		l.Hostname = host
+		for _, path := range rule.HTTP.Paths {
+
+			route := model.HTTPRoute{}
+
+			switch *path.PathType {
+			case slim_networkingv1.PathTypeExact:
+				route.PathMatch.Exact = path.Path
+			case slim_networkingv1.PathTypePrefix:
+				route.PathMatch.Prefix = path.Path
+			case slim_networkingv1.PathTypeImplementationSpecific:
+				route.PathMatch.Regex = path.Path
+			}
+
+			backend := model.Backend{
+				Name:      path.Backend.Service.Name,
+				Namespace: ing.Namespace,
+			}
+			if path.Backend.Service != nil {
+				backend.Port = &model.BackendPort{}
+				if path.Backend.Service.Port.Name != "" {
+					backend.Port.Name = path.Backend.Service.Port.Name
+				}
+				if path.Backend.Service.Port.Number != 0 {
+					backend.Port.Port = uint32(path.Backend.Service.Port.Number)
+				}
+			}
+			route.Backends = append(route.Backends, backend)
+			l.Routes = append(l.Routes, route)
+		}
+
+		insecureListenerMap[host] = l
+	}
+
+	secureListenerMap := make(map[string]model.HTTPListener)
+
+	// First, we check for TLS config, and set them up with Listeners to return.
+	for _, tlsConfig := range ing.Spec.TLS {
+
+		for _, host := range tlsConfig.Hosts {
+
+			var l model.HTTPListener
```

Comment:
I think we can remove this variable definition, since we are using the `:=` operator while accessing the `secureListenerMap` map.